src/functions/create-session/handler.ts

```diff
@@ -0,0 +1,94 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import { authenticationService } from "src/database/services";
+import { generateToken, parseBasicAuth, parseBearerAuth } from "@libs/auth-utils";
+import authenticationSchema from "src/schema/authentication";
+import { generateExpiryTimestamp } from "@libs/date-utils";
+
+/**
+ * Lambda for creating Authentication session
+ *
+ * @param event event
+ */
+const createAuthenticationSession: ValidatedEventAPIGatewayProxyEvent<typeof authenticationSchema> = async (event) => {
+  const { headers } = event;
+  const { Authorization, authorization } = headers;
+  const authHeader = Authorization || authorization;
+  const isBasicAuth = authHeader?.toLowerCase()?.startsWith("basic");
+  const isBearerAuth = authHeader?.toLowerCase()?.startsWith("bearer");
+
+  if (!isBearerAuth && !isBasicAuth) {
+    return {
+      statusCode: 401,
+      body: "Unauthorized"
+    };
+
+  } else if (isBearerAuth) {
```

Comment:
IMO makes it easier to read if instead of having an else if block, this would just be its own if block

Reply:
Changed